server/src/main/java/org/elasticsearch/search/aggregations/bucket/histogram/DateIntervalWrapper.java

```diff
@@ -97,7 +97,7 @@ public String getPreferredName() {
     private DateHistogramInterval dateHistogramInterval;
     private IntervalTypeEnum intervalType = IntervalTypeEnum.NONE;
 
-    public static <T extends DateIntervalConsumer> void declareIntervalFields(ObjectParser<T, String> parser) {
+    public static <T extends DateIntervalConsumer<T>> void declareIntervalFields(ObjectParser<T, String> parser) {
```

Comment:
AGain, shouldn't this be `T extends DateIntervalConsumer<?>>`? Am I not understanding this properly?

Reply:
It's one of these super-confusing recursive types. I think it's correct.